src/server/lib/recaptcha.ts

```diff
@@ -0,0 +1,36 @@
+import { CaptchaErrors } from '@/shared/model/Errors';
+import { RecaptchaV2 } from 'express-recaptcha';
+import {
+  RecaptchaResponseV2,
+  RecaptchaResponseV3,
+} from 'express-recaptcha/dist/interfaces';
+import { ApiError } from '../models/Error';
+import { getConfiguration } from './getConfiguration';
+import { logger } from './logger';
+
+const {
+  googleRecaptcha: { secretKey, siteKey },
+} = getConfiguration();
+
+/**
+ * Throws a generic error if the recaptcha check has failed.
+ * @param recaptchaResponse
+ */
+export const checkRecaptchaError = (
+  recaptchaResponse: RecaptchaResponseV3 | RecaptchaResponseV2 | undefined,
+) => {
+  if (recaptchaResponse?.error) {
+    logger.error(
+      'Problem verifying recaptcha, error response: ',
+      recaptchaResponse.error,
+    );
+    throw new ApiError({
+      message: CaptchaErrors.GENERIC,
+      status: 400,
+    });
+  }
+};
```

Comment:
can we add this to the middleware so we don't have to call checkRecaptchaError independently of adding the middleware for each route?

Reply:
hmm, I initially thought about adding it as middleware. the thing was that in each route we catch the thrown error and return the error to the rendered page like this: https://github.com/guardian/gateway/blob/3ccfa267589e4f0a8f6d095c8208109691557c97/src/server/routes/verifyEmail.ts#L118 I can't think of a clean way of doing the check outside of this try/catch block without shifting a lot of stuff around. I could be missing something here about how we could do it cleanly though?

Reply:
Maybe we can do something similar to how csrf errors are handled

Reply:
ah I think I see what you mean - will get back to you soon

Reply:
after pairing on this @QuarpT  and I moved reCAPTCHA error handling into its own query param called: `recaptchaError` which is caught in `errorHandler.ts` here: https://github.com/guardian/gateway/pull/1176/files#diff-ab4263f049cc27297065729715aa9cc12772fd4786a620419d124c0ace700d9fR36

To protect a route using this middleware you now need to use the exported array on the route: https://github.com/guardian/gateway/pull/1176/files#diff-1a6c97bd62b9062b06450a4be0bc71b232e70540be8dbbde0d6e194bd7901157R43